python/ad-hoc-ingest/ad_hoc_ingest.py

```diff
@@ -0,0 +1,293 @@
+import uuid
+from datetime import datetime
+from pathlib import Path, PureWindowsPath, PurePosixPath
+
+import pandas
+import pandas as pd
+from botocore.exceptions import ClientError
+from moto.utilities.utils import str2bool
+
+import aws_interactions
+import dataset_validator
+import discovery_client
+from dataset_validator import Js8Validator
+
+
+def build_argument_parser():
+    parser = argparse.ArgumentParser(
+        description="Process an input file to schedule corresponding ingests ",
+        add_help=False
+    )
+    parser.add_argument(
+        "-i", "--input",
+        required=True,
+        help="A CSV file containing details of the records to ingest, it must have columns (catRef, fileName, checksum)"
+    )
+    parser.add_argument(
+        "-e", "--environment",
+        help="Environment where the ingest is taking place (e.g. intg or prd)",
+        default="intg"
+    )
+    parser.add_argument(
+        "-d", "--dry_run",
+        nargs="?",
+        const=True,
+        type=str2bool,
+        help="Value of 'True' indicates that the tool will only validate inputs, without actually running an ingest",
+        default=False
+    )
+    parser.add_argument(
+        "-o", "--output",
+        help="Name of the folder to store a CSV file representing generated metadata for this ingest",
+        default=tempfile.gettempdir()
+    )
+    return parser
+
+# Validations of the parameters passed to the script.
+def validate_arguments(args):
+    input_file_path = Path(args.input)
+    if not (input_file_path.exists() and input_file_path.is_file()):
+        raise Exception(f"The input file [{input_file_path}] does not exist or it is not a valid file\n")
+
+    output_metadata_folder = Path(args.output)
+    if not (output_metadata_folder.exists() and output_metadata_folder.is_dir()):
+        raise Exception(f"The output metadata location [{output_metadata_folder}] does not exist or it is not a valid folder\n")
+
+    return True
+
+def create_metadata(row, args):
+    catalog_ref = row["catRef"].strip()
+    file_path = row["fileName"].strip()
+    collection_info = discovery_client.get_title_and_description(catalog_ref)
+    former_references = discovery_client.get_former_references(collection_info.identifier)
+
+    if not collection_info.title and not collection_info.description:
+        raise Exception(f"Title and Description both are empty for '{catalog_ref}', unable to proceed with this record")
+
+    description_to_use = collection_info.title if collection_info.title is not None else collection_info.description
+    series = row["catRef"].split("/")[0].strip()
+    metadata = {
+        "Series": series,
+        "UUID": str(uuid.uuid4()),
+        "fileId": str(uuid.uuid4()),
+        "description": description_to_use,
+        "Filename": get_filename_from_cross_platform_path(file_path),
+        "FileReference": catalog_ref.removeprefix(f"{series}/"),
+        "ClientSideOriginalFilepath": file_path,
```

Comment:
Couldn't you add "formerRefDept", "formerRefTNA", "checksum_sha256" and "checksum_md5" here directly, with a value of ""?

Reply:
Doesn't make much difference to code. in fact by not initialising default, there won't be accidents if someone changes the lines further down